### PR TITLE
re: migrate ReactCookieJarContainerTest to Kotlin

### PR DESCRIPTION
**Pull request by pk-218**
## Summary:
This PR migrates ReactCookieJarContainerTest.java to Kotlin as a part of the issue #37708 

## Changelog:
[INTERNAL] [CHANGED] - Migrate ReactCookieJarContainerTest.java to Kotlin

## Test Plan:
1. Run ./gradlew :packages:react-native:ReactAndroid:test.
2. All tests should pass.

### CODE REVIEW

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/modules/network/ReactCookieJarContainerTest.kt [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+import org.robolectric.RobolectricTestRunner
+
+/** Tests for {@link NetworkingModule}. */
+@PrepareForTest(ReactCookieJarContainer::class)
+@RunWith(RobolectricTestRunner::class)
+@PowerMockIgnore("org.mockito.*", "org.robolectric.*", "androidx.*", "android.*")
+class ReactCookieJarContainerTest {
+
+  @Test
+  @Throws(Exception::class)
```
Comment (rshest) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/modules/network/ReactCookieJarContainerTest.kt:32]: This is not needed, and can be removed (in other places as well).

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/modules/network/ReactCookieJarContainerTest.kt [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+import okhttp3.Cookie
+import okhttp3.CookieJar
+import okhttp3.HttpUrl
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito.any
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.`when`
```
Comment (rshest) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/modules/network/ReactCookieJarContainerTest.kt:20]: You can do `as whenever` here, and then you won't need to use backticks in the tests.

**Comment by rshest** [comment]
@pk-218 

Thanks for working on this one!

Sending back to you, though - the tests that you converted are failing, [see here](https://app.circleci.com/pipelines/github/facebook/react-native/25597/workflows/324316a7-e70f-46b1-8a30-f26f05e9a575/jobs/725941/parallel-runs/0/steps/0-109):

image

Also, I've left a couple of comments about minor stylistic issues, would appreciate if you address them as well :)

**Review by rshest — CHANGES_REQUESTED** [review]
See my comment above.

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/modules/network/ReactCookieJarContainerTest.kt [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  fun testMissingJar() {
+    val jarContainer: ReactCookieJarContainer = mock(ReactCookieJarContainer::class.java)
+    assertThat(jarContainer.loadForRequest("http://example.com".toHttpUrl()).size).isEqualTo(0)
+  }
+
+  @Test
+  @Throws(Exception::class)
+  fun testEmptyCookies() {
+    val jarContainer: ReactCookieJarContainer = mock(ReactCookieJarContainer::class.java)
+    val cookies: List<Cookie> = listOf<Cookie>()
```
Comment (cortinico) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/modules/network/ReactCookieJarContainerTest.kt:42]:
```suggestion
    val cookies: List<Cookie> = emptyList()
```

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/modules/network/ReactCookieJarContainerTest.kt [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    assertThat(jarContainer.loadForRequest("http://example.com".toHttpUrl()).size).isEqualTo(0)
+  }
+
+  @Test
+  @Throws(Exception::class)
+  fun testValidCookies() {
+    val jarContainer = ReactCookieJarContainer()
+    val cookieJar: CookieJar = mock(CookieJar::class.java)
+    jarContainer.setCookieJar(cookieJar)
+    val cookies: MutableList<Cookie> = mutableListOf<Cookie>()
```
Comment (cortinico) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/modules/network/ReactCookieJarContainerTest.kt:53]:
```suggestion
    val cookies: List<Cookie> = emptyList()
```

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/modules/network/ReactCookieJarContainerTest.kt [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    assertThat(jarContainer.loadForRequest("http://example.com".toHttpUrl()).size).isEqualTo(1)
+  }
+
+  @Test
+  @Throws(Exception::class)
+  fun testInvalidCookies() {
+    val jarContainer = ReactCookieJarContainer()
+    val cookieJar: CookieJar = mock(CookieJar::class.java)
+    jarContainer.setCookieJar(cookieJar)
+    val cookies: MutableList<Cookie> = mutableListOf<Cookie>()
```
Comment (cortinico) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/modules/network/ReactCookieJarContainerTest.kt:65]: same as above

**Review by cortinico — CHANGES_REQUESTED** [review]
Left a couple of minor nits, thanks for sending this @pk-218

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/modules/network/ReactCookieJarContainerTest.kt (hunk 4 above)
Comment (cortinico) [reply packages/react-native/ReactAndroid/src/test/java/com/facebook/react/modules/network/ReactCookieJarContainerTest.kt:53]: Nope it's fine as you wrote initially, my bad

**Comment by rshest** [comment]
Hey @pk-218, are you planning to follow up on this, or should we perhaps close it?

**Review by cortinico — APPROVED** [review]

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/modules/network/ReactCookieJarContainerTest.kt [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+  }
+
+  @Test
+  fun testValidCookies() {
+    val jarContainer = ReactCookieJarContainer()
+    val cookieJar: CookieJar = mock(CookieJar::class.java)
+    jarContainer.setCookieJar(cookieJar)
+    val cookies: MutableList<Cookie> = mutableListOf()
+    cookies.add(Cookie.Builder().name("valid").value("valid value").domain("domain").build())
+    whenever(cookieJar.loadForRequest(httpUrl)).thenReturn(cookies)
```
Comment (rshest) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/modules/network/ReactCookieJarContainerTest.kt:52]: Why `httpUrl` and not `any(HttpUrl::class.java)`?

Comment (pk-218) [reply packages/react-native/ReactAndroid/src/test/java/com/facebook/react/modules/network/ReactCookieJarContainerTest.kt:52]: Using an example string was the only was I could resolve the exceptions. This is a workaround. Please suggest a better approach

**Pull request closed without merging**